src/adapters/openx.js

```diff
@@ -1,130 +1,228 @@
- * @param {string} options.pageURL - Current page URL to send with bid request
- * @param {string} options.refererURL - Referer URL to send with bid request
- *
- * @returns {{callBids: _callBids}}
- * @constructor
- */
-var OpenxAdapter = function OpenxAdapter(options) {
-
-  var opts = options || {};
-  var scriptUrl;
-  var bids;
-
-  function _callBids(params) {
-    bids = params.bids || [];
-    for (var i = 0; i < bids.length; i++) {
-      var bid = bids[i];
-
-      //load page options from bid request
-      if (bid.params.pageURL) {
-        opts.pageURL = bid.params.pageURL;
-      }
+const bidfactory = require('../bidfactory.js');
+const bidmanager = require('../bidmanager.js');
+const adloader = require('../adloader');
+const CONSTANTS = require('../constants.json');
+const utils = require('../utils.js');
+
+const OpenxAdapter = function OpenxAdapter() {
+  const BIDDER_CODE = 'openx';
+  const BIDDER_CONFIG = 'hb_pb';
+  let startTime;
+
+  let pdNode = null;
+
+  $$PREBID_GLOBAL$$.oxARJResponse = function (oxResponseObj) {
+    let adUnits = oxResponseObj.ads.ad;
+    if (oxResponseObj.ads && oxResponseObj.ads.pixels) {
+      makePDCall(oxResponseObj.ads.pixels);
+    }
 
-      if (bid.params.refererURL) {
-        opts.refererURL = bid.params.refererURL;
-      }
+    if (!adUnits) {
+      adUnits = [];
+    }
 
-      if (bid.params.jstag_url) {
-        scriptUrl = bid.params.jstag_url;
+    var bids = $$PREBID_GLOBAL$$._bidsRequested.find(bidSet => bidSet.bidderCode === 'openx').bids;
+    for (let i = 0; i < bids.length; i++) {
+      let bid = bids[i];
+      let auid = null;
+      let adUnit = null;
+      // find the adunit in the response
+      for (let j = 0; j < adUnits.length; j++) {
+        adUnit = adUnits[j];
+        if (String(bid.params.unit) === String(adUnit.adunitid) && adUnitHasValidSizeFromBid(adUnit,bid) && !adUnit.used) {
+          auid = adUnit.adunitid;
+          break;
+        }
       }
 
-      if (bid.params.pgid) {
-        opts.pgid = bid.params.pgid;
+      let beaconParams = {
+        bd: +(new Date()) - startTime,
+        br: '0', // maybe 0, t, or p
+        bt: $$PREBID_GLOBAL$$.cbTimeout || $$PREBID_GLOBAL$$.bidderTimeout , // For the timeout per bid request
+        bs: window.location.hostname
+      };
+
+      // no fill :(
+      if (!auid) {
+        addBidResponse(null, bid);
+        continue;
       }
+      adUnit.used = true;
+
+      if (adUnit.pub_rev) {
+        beaconParams.br = beaconParams.bt < beaconParams.bd ? 't' : 'p';
+        beaconParams.bp = adUnit.pub_rev;
+        beaconParams.ts = adUnit.ts;
+        addBidResponse(adUnit, bid);
+      } 
+      buildBoPixel(adUnit.creative[0], beaconParams);
+    }
+  };
+
+  function getViewportDimensions(isIfr) {
+    var width,
+      height,
+      tWin = window,
+      tDoc = document,
+      docEl = tDoc.documentElement,
+      body;
+
+    if (isIfr) {
+      tWin = window.top;
```

Comment:
The window.top here is already checked below at line 195, I think it should be fine here without a try/catch?